Backport pipeline improvements for EL7/8/MicroOS

andypitcher opened the pull request:
Improve pipeline and scripts for EL7, EL8 and MicroOS.

- Remove dapper
- Add sign --dry-run
- Add repo-metadata script

andypitcher committed:
Update EL7's drone and scripts based on EL9

macedogm committed:
Create and upload repo metadata

Signed-off-by: Guilherme Macedo <guilherme@gmacedo.com>

macedogm commented on policy/centos7/scripts/build:
```diff
@@ -4,6 +4,9 @@ set -e -x
 . ./scripts/version
 
+yum install -y epel-release
+yum install -y container-selinux selinux-policy-devel yum-utils rpm-build # git
```
```suggestion
yum install -y container-selinux selinux-policy-devel yum-utils rpm-build
```

macedogm commented on policy/centos7/scripts/upload-repo:
```diff
@@ -1,16 +1,26 @@
 
+# Install the awscli-v2 from AWS
+curl "https://awscli.amazonaws.com/awscli-exe-linux-$(uname -m).zip" -o "awscliv2.zip"
+unzip awscliv2.zip
```
```suggestion
unzip -q awscliv2.zip
```

macedogm replied:
To make the unzip output quite and remove noise from the CI logs.

macedogm reviewed (changes requested):
Can you please also remove the following Docker volume lines that were only needed when using Dapper's Docker-in-Docker build?

**Part 1**
https://github.com/rancher/rancher-selinux/blob/e449419dc13e6fc66d6db8dd1f36a744d77bda15/.drone.yml#L14-L16

**Part 2**
https://github.com/rancher/rancher-selinux/blob/e449419dc13e6fc66d6db8dd1f36a744d77bda15/.drone.yml#L95-L98

andypitcher committed:
El7 remove dapper/dry-run and add upload-repo

andypitcher committed:
Use createrepo_c command instead of createrepo symlink
As seen in EL7, the symlink `/usr/bin/createrepo` isn't created by default during createrepo_c installation, so the `createrepo` command fails. As of now, we will use createrepo_c command in all distro.

andypitcher committed:
Merge branch 'main' into backport-pipeline-improvements

andypitcher committed:
EL8 improvements

andypitcher committed:
MicroOS improvements
- Remove dapper and use opensuse/tumbleweed
- Create/Modify scripts (use zypper)

macedogm reviewed (changes requested):
Some further changes (that I can't directly point in the PR):

1. In:
https://github.com/rancher/rancher-selinux/blob/686bc8593b812f06b53d1c4cec2cefb6dbe8c7c1/.drone.yml#L167

Need to match the same change done for EL7 -> `- "policy/centos8/dist/**/*.rpm"`

2. In:
https://github.com/rancher/rancher-selinux/blob/686bc8593b812f06b53d1c4cec2cefb6dbe8c7c1/.drone.yml#L353

Need to match the same change done for EL7 -> `- "policy/microos/dist/**/*.rpm"`

andypitcher committed:
Update EL8/MicroOS files location

andypitcher commented:
@macedogm last [commit](https://github.com/rancher/rancher-selinux/pull/27/commits/9b76fbf5d020a4bf5d9fe756fb0adb259835abba) implements your last [recommendation](https://github.com/rancher/rancher-selinux/pull/27#pullrequestreview-1706086357)..
